### PR TITLE
fix(Core/DB): Brewspewer hardcoded conditions

### LINKED ISSUE #4650: [NPC] Gregan Brewspewer - Missing text option
#### Quest [The Videre Elixir](https://wowgaming.altervista.org/aowow/?quest=4041)
[Gregan Brewspewer](https://wowgaming.altervista.org/aowow/?npc=7775) is missing the option for get instructions on how to find [Evoroot](https://classic.wowhead.com/item=11242/evoroot#guides).


##### CURRENT BEHAVIOUR:
There is no option.

![Screenshot from 2021-02-26 00-21-58](https://user-images.githubusercontent.com/2512197/109275397-877a4080-77c9-11eb-8a73-9819f81d7f6f.png)



##### EXPECTED BLIZZLIKE BEHAVIOUR:
There should be an option for instructions.
https://www.youtube.com/watch?v=vv-xXpREsd0&t=42




##### Extra info
```
$ git log

commit ccca2dedfad536ba8a09c0cf428ce718567f1e54 (HEAD -> master, origin/master, origin/HEAD)                                                                                                                                                                                                                                                                                                                                                 
Author: T1ti <40864460+T1ti@users.noreply.github.com>
Date:   Fri Feb 19 19:27:51 2021 +0100

    fix(Scripts/MoltenCore): Garr abilities timers (#4568)
```

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/96762152-npc-gregan-brewspewer-missing-text-option?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by claudiodfc**
Continuation of #4844

## Changes Proposed:
- Removed hardcoded conditions from the core
- Implemented creatures with quest gossips and vendors using a White Chat balloon as shown below
- Creature must have a "Speak" Icon name for the upper to work on Creature Template

## Issues Addressed:

- Closes #4650 

## Tests Performed:

- Compiled and tested in Windows 10

## How to Test the Changes:

1. .go c 50145
2. Talk to the NPC and see that there are no gossip menu options
3. The Creature must also have this Icon when hovering:
image

4. .quest add 3909
5. Talk to the NPC and there are gossip menu options

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Sedatyf** [comment]
You made another PR?

**Comment by Sedatyf** [comment]
Alright, everything work as intended! ✔️ 
First, we have the gossip on hover, and when you speak to him, he just have a small text
After giving myself the quest, I had the blue question mark, and he has the gossip options and the buying option to buy bait's quest.
Finally, I spoke with several merchants in Stormwind. Hover tooltips were good as well as trading with them. Thank you for this @claudiodfc!

File: data/sql/updates/pending_db_world/rev_1623327182605839200.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+(15, 1802, 1, 0, 0, 47, 0, 3909, 8, 0, 0, 0, 0, '', '');
+
+-- Brewspewer has a White ballon icon when you over him regardless of his vendor flags
+UPDATE `creature_template` SET `IconName` = 'Speak' WHERE (`entry` = 7775);
+
+-- Removing his script name
+UPDATE `creature_template` SET `ScriptName` = '' WHERE (`entry` = 7775);
+
+-- This NPC has his npc flags overriden
+UPDATE `creature` SET `npcflag`='131' WHERE  `guid`=50145;
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1623327182605839200.sql:18]: missing new line at the end of file

image

File: src/server/game/Entities/Creature/Creature.h [hunk 2]
```diff
@@ -45,6 +45,8 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
 
     void DisappearAndDie();
 
+    bool isVendorWithIconSpeak() const;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Creature/Creature.h:48]: please consider using `[[nodiscard]]` where applicable. This looks a good candidate for it

**Comment by claudiodfc** [comment]
@FrancescoBorzi What's the review status? Should I change something?

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@claudiodfc sorry for the delay, approved. Good job and also thanks to @Sedatyf for testing this

**Pull request merged**